src/printer.js

```diff
@@ -1791,6 +1791,14 @@ function genericPrintNoParens(path, options, print, args) {
     case "JSXOpeningElement": {
       const n = path.getValue();
 
+      const nameHasComments =
+        n.name && n.name.comments && n.name.comments.length;
+
+      // Don't break self-closing elements with no attributes and no comments
+      if (n.selfClosing && (!n.attributes || !n.attributes.length) && !nameHasComments) {
```

Comment:
As far as I can tell from the next if-statement, `n.attributes` is always an array, so there’s no need to check if it exists.

Reply:
Yeah I was just following some previous code. I never know when properties that can be empty arrays can be undefined.